File: backend/dotnet/ticketflow/TicketFlow.TicketService.Client/Entities/Ticket.cs
using System;
using TicketFlow.TicketService.Client.Extensibility.Entities;

namespace TicketFlow.TicketService.Client.Entities
{
    internal class Ticket : ITicket
    {
        public Ticket(int id, int movieId, int row, int seat, int price)
        {
            Id = id;
            MovieId = movieId;
            Row = row;
            Seat = seat;
            Price = price;
        }

        public int Id { get; }

        public int MovieId { get; }

        public int Row { get; }

        public int Seat { get; }

        public int Price { get; }

        public IOrderedTicket Order(string buyerEmail)
        {
            if (string.IsNullOrEmpty(buyerEmail))
            {
                throw new ArgumentNullException();
            }

            return new OrderedTicket(Id, MovieId, Row, Seat, Price, buyerEmail);
        }
    }
}

File: backend/java/ticket_service/src/main/java/com/ticketflow/ticket_service/models/Ticket.java
package com.ticketflow.ticket_service.models;

import java.util.Optional;

public class Ticket {
    private Integer id;
    private Integer movieId;
    private String buyerEmail;
    private Integer row;
    private Integer seat;
    private Integer price;

    public Ticket(Integer id, Integer movieId, Optional<String> optionalBuyerEmail, Integer row, Integer seat, Integer price) {
        this.id = id;
        this.movieId = movieId;
        this.setBuyerEmail(optionalBuyerEmail);
        this.row = row;
        this.seat = seat;
        this.price = price;
    }

    public Integer getId() {
        return id;
    }

    public Integer getMovieId() {
        return movieId;
    }

    public void setMovieId(Integer movieId) {
        this.movieId = movieId;
    }

    public Optional<String> getBuyerEmail() {
        return Optional.ofNullable(buyerEmail);
    }

    public void setBuyerEmail(Optional<String> optionalBuyerEmail) {
        if (optionalBuyerEmail.isPresent()) {
            this.buyerEmail = optionalBuyerEmail.get();
        }
        else {
            this.buyerEmail = null;
        }
    }

    public Integer getRow() {
        return row;
    }

    public void setRow(Integer row) {
        this.row = row;
    }

    public Integer getSeat() {
        return seat;
    }

    public void setSeat(Integer seat) {
        this.seat = seat;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }
}